src/pytesmo/time_series/anomaly.py

```diff
@@ -87,19 +87,35 @@ def calc_anomaly(Ser,
     return anomaly
 
 
+def _index_units(year, month, day, unit="day", respect_leap_years=True) -> (np.array, int):
+    """Provide the indices for the specified unit type and the index lenth"""
+    if respect_leap_years:
+        args = month, day, year
+    else:
+        args = month, day
+
+    if unit == "day":
+        return doy(*args), 366
+    elif unit == "month":
+        return month, 12
+    else:
+        raise ValueError(f"Invalid unit: {unit}")
+
+
 def calc_climatology(Ser,
                      moving_avg_orig=5,
-                     moving_avg_clim=30,
+                     moving_avg_clim=35,
```

Comment:
For unit="month" 35 is not a good default value, this should probably be checked at the start of the function, something like
```
if unit == "month" and moving_avg_clim == 35:
    moving_avg_clim = <default value>
```
where the default value could be 1, 2, or 3.

Reply:
This should then also solve the failing tests

Reply:
`test_climatology_always_366_fill` and `test_climatology_interpolate_leapday` fail because we are now reindexing `clim_ser` in line 223, i.e. before the `moving_average()` and this leads somehow to a non-gapped climatology. I think these tests are obsolete as they are, but not sure how they should be updated

Reply:
For `test_climatology_always_366_fill` you can set `moving_avg_clim=1`. This way there will still be gaps in the climatology, I think. I would remove the other test actually, because the behaviour that it asserts does not make a lot of sense in my opinion.